Problem2.py

```diff
@@ -0,0 +1,61 @@
+"""
+#Given a sorted array of unknown length and a number to search for, return the index of the number in the array.
+# Accessing an element out of bounds throws exception. If the number occurs multiple times, return the index of any occurrence.
+# If it isn’t present, return -1.
+
+# we will expand our range by multyping high by factor of 2 and set low as high in each iteration. and find the low and high range
+
+// Time Complexity : O(log n)
+// Space Complexity : O(1)
+// Did this code successfully run on Leetcode : Didn't try
+// Any problem you faced while coding this :  Yes, in finding the Time Complexity  and the range of array
+ """
+class Solution:
+    def Search_Infinite(self,nums,target):
+        low=0   # intialise low as 0
+        high=2   # high as 2
+        try:     #The try block lets you test a block of code for errors. The except block lets you handle the error.
+            while nums[high]<target:   # if target is less than nums[high] we got to except part
+                low=high
+                high=high*2
+        except Exception as e:
+            try:        #
+                high=low+1
```

Comment:
This will make it linear.

Reply:
You don't have to worry about having the number in the last bucket because it's an infinite array.
Still you can refer https://github.com/devashishnyati/Binary-Search-1/blob/master/unkownlength.py